sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -389,12 +425,21 @@ public:
   void make_edge(node& src, node& dest);
 
   void print_graph(std::string path, bool verbose = false) const;
+
+  std::vector<node> get_nodes() const;
+  std::vector<node> get_root_nodes() const;
 };
 
 template<>
 class command_graph<graph_state::executable> {
 public:
-    command_graph() = delete;
+  command_graph() = delete;
+
+  void update_dynamic_parameter(dynamic_parameter dynamicParam,
+                                T newValue);
+  template<int Dimensions>
+  void update_nd_range(const std::vector<node>& nodes,
+                       nd_range<Dimensions> executionRange);
```

Comment:
Could you clarify what you mean by changing the CFG? I don't think so, if this could change the type of node completely if you used a different node type, e.g. changed a parallel_for to a memcopy. Even if you changed the parallel_for function, that would meaning a different underlying pi/ur kernel object. This functionality couldn't be represented natively in any SYCL-2020 backend I'm aware of, so would require emulation by creating the whole backend command-buffer.

Reply:
I meant to change the parallel_for function. It's not a requirement, but a discussion, and thanks for the sharing.